### PR TITLE
fix: approvals

### PR DESCRIPTION
**Pull request by janndriessen**
⚠️ DO NOT MERGE WITHOUT REVERTING THE LAST COMMIT ⚠️ 

## **Summary of Changes**

* Fixes approval issues where the approval state would not be recognized (after success), not updating the trade button

## Docs
* https://1.x.wagmi.sh/react/hooks/useContractRead

## Testing

* [e0dec2d](https://github.com/IndexCoop/index-app/pull/1063/commits/e0dec2d3ae528637f33984992e24d1d235ab9a2b) is necessary for testing to work but will have to be reverted before merging! ⚠️ 
* Start hardhat node: `npm run hardhat`
* Fund one of the testing accounts (address specified in .env.local): `npm run hardhat:fund`
* Connect to `localhost`
* Use USDC as currency e.g. USDC-ETH2X
* Input amount e.g. 10 USDC
* Test approving (and state changes of button) 🧪 🔬 

In case you restart the hardhat node and receive nonce issues with the test account, follow these instructions to reset the account data in metamask: https://support.metamask.io/hc/en-us/articles/360015488891-How-to-clear-your-account-activity-reset-account. Make sure that the test account is selected!

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ module.exports = {
   networks: {
     hardhat: {
       forking: {
+        chainId: 31337,
```
Comment (janndriessen) [inline hardhat.config.js:10]: otherwise, can not connect to localhost

File: src/lib/hooks/use-best-quote/utils/flashmint.ts [hunk 2]
```diff
@@ -194,7 +193,7 @@ export async function getFlashMintQuote(
       slippage,
       chainId,
       provider,
-      signer
+      signer,
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/flashmint.ts:196]: all these changes will have to be reverted before merging

File: src/lib/utils/tokens.ts [hunk 3]
```diff
@@ -30,6 +30,8 @@ export function getAddressForToken(
   token: Token,
   chainId: number | undefined,
 ): string | undefined {
+  if (token.symbol === IndexCoopEthereum2xIndex.symbol) return token.address
+  if (token.symbol === USDC.symbol) return token.address
```
[the remark this answers is not in the document]
Comment (janndriessen) [reply src/lib/utils/tokens.ts:34]: have to see how we might handle this in the future to make things testable - as chainId of a local fork will never be that of mainnet, I guess.

**Review by 0xonramp — APPROVED** [review]
LGTM

**Pull request merged**